libraries/plugins/webserver/webserver_plugin.cpp

```diff
@@ -134,19 +138,28 @@ class webserver_plugin_impl
       plugins::json_rpc::json_rpc_plugin* api;
 };
 
-webserver_plugin::webserver_plugin() : _my( new webserver_plugin_impl() ) {}
-webserver_plugin::~webserver_plugin(){}
+webserver_plugin::webserver_plugin() {}
+webserver_plugin::~webserver_plugin() {}
 
 void webserver_plugin::set_program_options( options_description&, options_description& cfg )
 {
    cfg.add_options()
-         ("webserver-http-endpoint", bpo::value< string >(), "The local IP and port to listen for incoming http connections.")
-         ("webserver-ws-endpoint", bpo::value< string >(), "The local IP and port to listen for incoming websocket connections.")
-         ;
+      ("webserver-http-endpoint", bpo::value< string >(), "The local IP and port to listen for incoming http connections.")
+      ("webserver-ws-endpoint", bpo::value< string >(), "The local IP and port to listen for incoming websocket connections.")
+      (WEBSERVER_THREAD_POOL_SIZE_OPTION, bpo::value<thread_pool_size_t>()->default_value(WEBSERVER_DEFAULT_THREAD_POOL_SIZE),
+      "Number of threads used to handle queries. Default: " BOOST_PP_STRINGIZE(WEBSERVER_DEFAULT_THREAD_POOL_SIZE) ".")
+      ;
 }
 
 void webserver_plugin::plugin_initialize( const variables_map& options )
 {
+   FC_ASSERT(options.count(WEBSERVER_THREAD_POOL_SIZE_OPTION) != 0, WEBSERVER_THREAD_POOL_SIZE_OPTION "is required!");
+   auto thread_pool_size = options.at(WEBSERVER_THREAD_POOL_SIZE_OPTION).as<thread_pool_size_t>();
+   if (thread_pool_size <= 0)
```

Comment:
The catch block in main.cpp should have a catch for fc::exception that prints the error message.